cdci_data_analysis/analysis/parameters.py

```diff
@@ -674,85 +673,42 @@ def _set_angle(self, value, units):
 #
 
 
+class Energy(Float):
+    def __init__(self, value=None, E_units=None, name=None, check_value=None):
 
-class SpectralBoundary(Parameter):
-    def __init__(self,value=None,E_units='keV',name=None):
+        if E_units is None:
+            E_units = 'keV'
 
-        _allowed_units = ['keV','eV','MeV','GeV','TeV','Hz','MHz','GHz']
-
-        #wtform_dict = {'keV': FloatField}
-
-        super(SpectralBoundary, self).__init__(value=value,
-                                   units=E_units,
-                                   check_value=self.check_energy_value,
-                                   name=name,
-                                   allowed_units=_allowed_units)
-                                   #wtform_dict=wtform_dict)
-
-
-
-
-    @staticmethod
-    def check_energy_value(value, units=None,name=None):
-        #print('check type of ',name,'value', value, 'type',type(value))
-
-
-        try:
-            value=ast.literal_eval(value)
-        except:
-            pass
-
-        if type(value)==int or type(value)==np.int:
-            pass
-        elif type(value)==float or type(value)==np.float:
-            pass
-        else:
-            raise RuntimeError('type of ',name,'not valid',type(value))
-
-
-class Energy(Parameter):
-    def __init__(self,value=None,E_units=None,name=None):
-
-        _allowed_units = ['keV','eV','MeV','GeV','TeV']
+        if check_value is None:
+            check_value = self.check_float_value
 
+        _allowed_units = ['keV', 'eV', 'MeV', 'GeV', 'TeV', 'Hz', 'MHz', 'GHz']
         #wtform_dict = {'keV': FloatField}
 
         super(Energy, self).__init__(value=value,
-                                   units=E_units,
-                                   check_value=self.check_energy_value,
-                                   name=name,
-                                   allowed_units=_allowed_units)
+                                     units=E_units,
+                                     default_units='keV',
+                                     check_value=check_value,
+                                     name=name,
+                                     allowed_units=_allowed_units)
                                    #wtform_dict=wtform_dict)
 
 
+class SpectralBoundary(Energy):
+    def __init__(self, value=None, E_units='keV', name=None):
```

Comment:
After the last changes, there is very little difference between this class and `Energy, but this class is used a lot within the plugin.

Otherwise, I would just simply suggest to remove it

Reply:
Since it's already used, removing is challenging.

But it can be just 
```suggestion
class SpectralBoundary(Energy):
    pass
```